sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -136,6 +136,29 @@ case class UnresolvedTableValuedFunction(
   override lazy val resolved = false
 }
 
+/**
+ * A table-valued command, e.g.
+ * {{{
+ *   select tableName from command("show tables");
```

Comment:
We cannot add a new built-in function for this purpose instead? The implementation will be simpler than the current syntax-based approach, I think.

Reply:
First, this PR references the table-valued-function. So, I call the syntax is table-valued-command.
Second, Users want use filter clause, group by, ... on table-valued-command. In fact, This PR adopt the syntax-based approach is much simpler than built-in function.

Reply:
Yes. This implementation can make these command easy to use:
1. Save result to table.
2. Query the specified column.
3. Filter / Group by / Order by the specified column.

This is the syntax of Teradata:
```sql
SELECT  tbl.DatabaseName,
        tbl.TableName,
        SUM(spc.CurrentPerm)/1024.00 as TableSize
FROM    DBC.TablesV tbl
JOIN    DBC.TableSize spc
ON  tbl.DatabaseName = spc.DatabaseName
AND tbl.TableName = spc.TableName
WHERE   tbl.DatabaseName NOT IN ('All', 'Crashdumps', 'DBC', 'dbcmngr', 
        'Default', 'External_AP', 'EXTUSER', 'LockLogShredder', 'PUBLIC',
        'Sys_Calendar', 'SysAdmin', 'SYSBAR', 'SYSJDBC', 'SYSLIB', 
        'SystemFe', 'SYSUDTLIB', 'SYSUIF', 'TD_SERVER_DB',  'TDStats',
        'TD_SYSGPL', 'TD_SYSXML', 'TDMaps', 'TDPUSER', 'TDQCD',
        'tdwm',  'SQLJ', 'TD_SYSFNLIB',  'SYSSPATIAL')
AND TableKind = 'T'
GROUP BY    tbl.DatabaseName,
            tbl.TableName
ORDER BY TableSize DESC;
```